spec/routes_spec.rb

```diff
@@ -74,14 +74,18 @@ def fp_message
               allow(subject).to receive(:redirect_to).with(:controller => 'dashboard', :action => 'auth_error').and_raise(MiqException::RbacPrivilegeException)
               allow(subject).to receive(:add_flash).with("The user is not authorized for this task or item.", :error).and_raise(MiqException::RbacPrivilegeException)
 
-              expect do
-                subject.send(:check_privileges)
-                subject.send(action)
-              rescue MiqException::RbacPrivilegeException
+              begin
+                expect do
+                  subject.send(:check_privileges)
+                  subject.send(action)
+                rescue MiqException::RbacPrivilegeException
+                  raise
+                end.to raise_error(MiqException::RbacPrivilegeException)
+              rescue RSpec::Expectations::ExpectationNotMetError
+                # Append a custom error message
+                $!.message << "\n\n" + error_message unless fp
```

Comment:
I assume `$!` is the original error?  Can it also be pulled from the `ExpectationNotMetError` with something like:
```suggestion
              rescue RSpec::Expectations::ExpectationNotMetError => err
                # Append a custom error message
                "${err.cause.message}\n\n#{error_message}" unless fp
```